app/app.go

```diff
@@ -704,8 +741,38 @@ func NewApp(
 		tokenfactory.NewAppModule(app.TokenFactoryKeeper, app.AccountKeeper, app.BankKeeper, app.GetSubspace(tokenfactorytypes.ModuleName)),
 		poamodule.NewAppModule(appCodec, app.POAKeeper),
 		// sdk
-		crisis.NewAppModule(app.CrisisKeeper, skipGenesisInvariants, app.GetSubspace(crisistypes.ModuleName)), // always be last to make sure that it checks for all invariants and not only part of them
+		crisis.NewAppModule(app.CrisisKeeper, skipGenesisInvariants, app.GetSubspace(crisistypes.ModuleName)),
 		manifest.NewAppModule(appCodec, app.ManifestKeeper, app.MintKeeper),
+		wasm.NewAppModule(appCodec, &app.WasmKeeper, app.StakingKeeper, app.AccountKeeper, app.BankKeeper, app.MsgServiceRouter(), app.GetSubspace(wasmtypes.ModuleName)),
+	)
+
+	// Add wasm module to the app module manager
+	app.ModuleManager.Modules[wasmtypes.ModuleName] = wasm.NewAppModule(
+		appCodec,
+		&app.WasmKeeper,
+		app.StakingKeeper,
+		app.AccountKeeper,
+		app.BankKeeper,
+		app.MsgServiceRouter(),
+		app.GetSubspace(wasmtypes.ModuleName),
+	)
+
+	// Add wasm to begin blockers
+	app.ModuleManager.OrderBeginBlockers = append(
+		app.ModuleManager.OrderBeginBlockers,
+		wasmtypes.ModuleName,
+	)
+
+	// Add wasm to end blockers
+	app.ModuleManager.OrderEndBlockers = append(
+		app.ModuleManager.OrderEndBlockers,
+		wasmtypes.ModuleName,
+	)
+
+	// Add wasm to init genesis
+	app.ModuleManager.OrderInitGenesis = append(
+		app.ModuleManager.OrderInitGenesis,
+		wasmtypes.ModuleName,
 	)
```

Comment:
I have never seen anyone do this before, not sure what hack this is working around but shouldn't be done like this

Reply:
Fixed in e55694c6dd41a7fae2bb450f10390463403e7739